src/wmtk/simplex/SimplexCollection.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <vector>
+#include <wmtk/Mesh.hpp>
+#include "Simplex.hpp"
+#include "internal/SimplexEqualFunctor.hpp"
+#include "internal/SimplexLessFunctor.hpp"
+
+namespace wmtk::simplex {
+class SimplexCollection
+{
+public:
+    SimplexCollection(const Mesh& mesh)
+        : m_mesh{mesh}
+        , m_simplex_is_less(mesh)
+        , m_simplex_is_equal(mesh)
+    {}
+
+    /**
+     * @brief Return const reference to the simplex vector.
+     */
+    const std::vector<Simplex>& simplex_vector() const { return m_simplices; }
+    /**
+     * @brief Return vector of all simplices of the requested type.
+     */
+    std::vector<Simplex> simplex_vector(const PrimitiveType& ptype) const;
```

Comment:
this could be `get_by_order` or something?

Reply:
not sure about the name but we can definitely change it